### PR TITLE
file: New package

### PR DESCRIPTION
**Pull request by pfmoore**

### CODE REVIEW

File: mingw-w64-file/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Maintainer: Alexey Pavlov <Alexpux@gmail.com>
+# Contributor: Paul Moore <p.f.moore@gmail.com>
+
```
Comment (mingwandroid) [inline mingw-w64-file/PKGBUILD]: You should list yourself as maintainer and not blame :-) alexey for any of it, IMHO.

File: mingw-w64-file/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+_realname=file
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=5.22
+pkgrel=1
+pkgdesc='The file command is "a file type guesser", that is, a command-line tool that tells you in words what kind of data a file contains'
```
Comment (mingwandroid) [inline mingw-w64-file/PKGBUILD]: Please add (mingw-w64) and perhaps otherwise shorten the description

**Comment by pfmoore** [comment]
Changes made as requested. Sorry about the "Maintainer" thing, I copied it from another PKGBUILD without thinking too much. I accept any and all blame in future :-)

**Comment by mingwandroid** [comment]
Hate to be a pain again Paul, but can you squash the two commits so our history is cleaner?

**Comment by pfmoore** [comment]
Sure. I wasn't sure what the policy was here. Update incoming.

**Comment by elieux** [comment]
Builds fine here. Thank you, Paul, for your continuing work on these packages. I hope I'll be able to ditch GnuWin32 very soon.

File: mingw-w64-file/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Maintainer: Paul Moore <p.f.moore@gmail.com>
+
+_realname=file
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=5.22
+pkgrel=1
+pkgdesc='Determine the type of a file from its contents (mingw-w64)'
+groups=("${MINGW_PACKAGE_PREFIX}")
```
Comment (Alexpux) [inline mingw-w64-file/PKGBUILD]: remove ````groups```` field please

File: mingw-w64-file/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+arch=('any')
+url='http://www.darwinsys.com/file/'
+
```
Comment (Alexpux) [inline mingw-w64-file/PKGBUILD]: And empty lines remove before ````prepare````

File: mingw-w64-file/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+prepare() {
+  cd $srcdir/${_realname}-${pkgver}
+}
+
+build() {
+  cd $srcdir/${_realname}-${pkgver}
+  ./configure --prefix=${MINGW_PREFIX} \
+    --build=${MINGW_CHOST} \
+    --host=${MINGW_CHOST} \
+    --target=${MINGW_CHOST} \
```
Comment (Alexpux) [inline mingw-w64-file/PKGBUILD]: Remove last slash as this is last argument.

**Comment by Alexpux** [comment]
Also what about paths relocations? Does ````file```` read it's files from /mingw*/share/* folder?

**Comment by elieux** [comment]
> Does `file` read it's files from `/mingw/share/` folder?

I tried it and it does (when I delete `share/misc/magic.mgc`, `file` stops giving results).

**Comment by Alexpux** [comment]
It works for you because you run ````file```` with your hardcoded paths. Try install your file package on different computer with different msys installation folder location.

File: mingw-w64-file/PKGBUILD (hunk 5 above)
Comment (elieux) [inline mingw-w64-file/PKGBUILD]: (Not an issue, but I'm curious.) Is `--target` supposed to be used here? Isn't that option only for compilers and such?

Comment (Alexpux) [inline mingw-w64-file/PKGBUILD]: You can use all 3 (build,host,target) options in 90% packages without the issue.

**Comment by pfmoore** [comment]
I've fixed the typos in the file. As regards relocatability, I think it works. I checked the code and it has the standard relocation handling in there (magic.c line 179 calls GetModuleFileNameA to look for magic.mgc relative to the dll location).

I'll recheck with a different install location, but did you actually experience a problem, or were you just concerned that there wasn't a relocation patch?

**Pull request merged**